[Bug] IceGenericError when using Metacello loading script with the last P11 VM

I have a big problem when I'm using a Metacello script since last P11 release + last VM.
This problem **appears for all projects**.

With the previous VM : no problem. It's working with a previous version.

![image](https://github.com/pharo-project/pharo/assets/49183340/b05658ee-d4f0-4ea8-8c21-3870c71c3d1d)

Create a new P11 image, last VM and do this script : 

```Smalltalk
Metacello new
   baseline: 'Molecule';
   repository: 'github://OpenSmock/Molecule';
   load.
```

My config : 
- Pharo 11 release (Pharo-11.0.0+build.689 64 bits)
- Windows 10 Pro
- VM version : CoInterpreter VMMaker-tonel.1 uuid: a8adc99c-b5b7-0d00-9966-921d0644bd08 May 10 2023
StackToRegisterMappingCogit VMMaker-tonel.1 uuid: a8adc99c-b5b7-0d00-9966-921d0644bd08 May 10 2023
v10.0.5 - Commit: 2757766f - Date: 2023-05-05 13:46:09 +0200

At image start there is this trace : 

```
[ERROR] 2023-10-09 16:09:20.000 sqMakeMemoryNotExecutableFromTo extracted/vm/src/win/sqWin32SpurAlloc.c:226):VirtualProtect(x,y,PAGE_EXECUTE_READWRITE): No error
```

And when I'm loading my baseline this is this trace :

```
←[31mNotFound: revspec 'alpha1' not found
←[0mIceLibgitErrorVisitor>>visitENOTFOUND:
LGit_GIT_ENOTFOUND>>acceptError:
[ :error | error acceptError: (IceLibgitErrorVisitor onContext: self) ] in IceLibgitRepository>>handleLibgitError: in Block: [ :error | error acceptError: (IceLibgitErrorV[..]
FullBlockClosure(BlockClosure)>>cull:
Context>>evaluateSignal:
Context>>handleSignal:
LGit_GIT_ENOTFOUND(Exception)>>signal
LGit_GIT_ENOTFOUND class(LGitCallReturnHandler class)>>signalWith:
LGitReturnCodeEnum>>handleLGitReturnCode
LGitRepository(LGitExternalObject)>>withReturnHandlerDo:
LGitRepository>>revparse:
[ | gitCommit gitId |
                gitId := (self repositoryHandle revparse: aName) object id.
                gitCommit := LGitCommit of: self repositoryHandle fromId: gitId.
                ^ self commitFromGitCommit: gitCommit ] in IceLibgitRepository>>revparse: in Block: [ | gitCommit gitId |...
FullBlockClosure(BlockClosure)>>on:do:
IceLibgitRepository>>handleLibgitError:
IceLibgitRepository>>revparse:
[ "If there is a single remote, we can look for the branch in this single remote."
                        self remotes size = 1
                                ifTrue: [ self remotes anyOne
                                                remoteBranchNamed: aName
                                                ifPresent: [ :found | ^ found ]
                                                ifAbsent: [ "Continue with the revparse" ] ].
                        self revparse: aName ] in IceLibgitRepository>>commitishNamed: in Block: [ "If there is a single remote, we can look fo[..]
UndefinedObject>>ifNil:ifNotNil:
[ | found |
                found := self repositoryHandle lookupLocalBranch: aString ifAbsent: [ nil ].
                ^ found ifNil: absentBlock ifNotNil: [ presentBlock value: (self basicNewBranchNamed: found name) ] ] in IceLibgitRepository>>branchNamed:ifPresent:ifAbsent: in Block: [ | found |...
FullBlockClosure(BlockClosure)>>on:do:
IceLibgitRepository>>handleLibgitError:
IceLibgitRepository>>branchNamed:ifPresent:ifAbsent:
IceLibgitRepository(IceRepository)>>branchNamed:ifAbsent:
IceLibgitRepository>>commitishNamed:
IceMetacelloRepositoryAdapter>>goferReferences
[ aRepository goferReferences asSortedCollection asArray ] in MetacelloGofer(Gofer)>>basicReferencesIn: in Block: [ aRepository goferReferences asSortedCollecti[..]
FullBlockClosure(BlockClosure)>>on:do:
MetacelloGofer(Gofer)>>basicReferencesIn:
[ :cache |
                        ^ cache at: aRepository put: (super basicReferencesIn: aRepository) ] in MetacelloGofer>>basicReferencesIn: in Block: [ :cache | ...
[:dict | | cache |
                        cache := dict at: cacheName ifAbsent: [].
                        cache ~~ nil
                                ifTrue: [ | value hasEntry |
                                        hasEntry := true.
                                        value := cache at: key ifAbsent: [ hasEntry := false ].
                                        hasEntry ifTrue: [ ^value ]]
                                ifFalse: [
                                        cache := cacheClass new.
                                        dict at: cacheName put: cache ].
                        ^aBlock value: cache ] in IceMetacelloPharoPlatform(MetacelloPlatform)>>stackCacheFor:cacheClass:at:doing: in Block: [:dict | | cache |...
[ ^ aBlock value: dict ] in IceMetacelloPharoPlatform(MetacelloPlatform)>>useStackCacheDuring:defaultDictionary: in Block: [ ^ aBlock value: dict ]
←[0m←[31mNotFound: revspec 'alpha1' not found
←[0mIceLibgitErrorVisitor>>visitENOTFOUND:
LGit_GIT_ENOTFOUND>>acceptError:
[ :error | error acceptError: (IceLibgitErrorVisitor onContext: self) ] in IceLibgitRepository>>handleLibgitError: in Block: [ :error | error acceptError: (IceLibgitErrorV[..]
FullBlockClosure(BlockClosure)>>cull:
Context>>evaluateSignal:
Context>>handleSignal:
LGit_GIT_ENOTFOUND(Exception)>>signal
LGit_GIT_ENOTFOUND class(LGitCallReturnHandler class)>>signalWith:
LGitReturnCodeEnum>>handleLGitReturnCode
LGitRepository(LGitExternalObject)>>withReturnHandlerDo:
LGitRepository>>revparse:
[ | gitCommit gitId |
                gitId := (self repositoryHandle revparse: aName) object id.
                gitCommit := LGitCommit of: self repositoryHandle fromId: gitId.
                ^ self commitFromGitCommit: gitCommit ] in IceLibgitRepository>>revparse: in Block: [ | gitCommit gitId |...
FullBlockClosure(BlockClosure)>>on:do:
IceLibgitRepository>>handleLibgitError:
IceLibgitRepository>>revparse:
[ "If there is a single remote, we can look for the branch in this single remote."
                        self remotes size = 1
                                ifTrue: [ self remotes anyOne
                                                remoteBranchNamed: aName
                                                ifPresent: [ :found | ^ found ]
                                                ifAbsent: [ "Continue with the revparse" ] ].
                        self revparse: aName ] in IceLibgitRepository>>commitishNamed: in Block: [ "If there is a single remote, we can look fo[..]
UndefinedObject>>ifNil:ifNotNil:
[ | found |
                found := self repositoryHandle lookupLocalBranch: aString ifAbsent: [ nil ].
                ^ found ifNil: absentBlock ifNotNil: [ presentBlock value: (self basicNewBranchNamed: found name) ] ] in IceLibgitRepository>>branchNamed:ifPresent:ifAbsent: in Block: [ | found |...
FullBlockClosure(BlockClosure)>>on:do:
IceLibgitRepository>>handleLibgitError:
IceLibgitRepository>>branchNamed:ifPresent:ifAbsent:
IceLibgitRepository(IceRepository)>>branchNamed:ifAbsent:
IceLibgitRepository>>commitishNamed:
IceMetacelloRepositoryAdapter>>goferReferences
[ aRepository goferReferences asSortedCollection asArray ] in MetacelloGofer(Gofer)>>basicReferencesIn: in Block: [ aRepository goferReferences asSortedCollecti[..]
FullBlockClosure(BlockClosure)>>on:do:
MetacelloGofer(Gofer)>>basicReferencesIn:
[ :cache |
                        ^ cache at: aRepository put: (super basicReferencesIn: aRepository) ] in MetacelloGofer>>basicReferencesIn: in Block: [ :cache | ...
[Error printing the compiledBlock in IceMetacelloPharoPlatform(MetacelloPlatform)>>stackCacheFor:cacheClass:at:doing: in Block: [:dict | | cache |...
[ ^ aBlock value: dict ] in IceMetacelloPharoPlatform(MetacelloPlatform)>>useStackCacheDuring:defaultDictionary: in Block: [ ^ aBlock value: dict ]
←[0m
```

And there is an another exception, the 'IceGenericError' is replaced by this one: 

![image](https://github.com/pharo-project/pharo/assets/49183340/3a5c9a53-6b6e-408d-b5d7-d2e9ec145c6d)

I think that **alpha1** is this project : 

```smalltalk
spec baseline: 'OnBloc' with: [ 
		spec repository: 'github://OpenSmock/OnBloc:alpha1/src'.
	].
```

1 - With a clean image + VM and DoIt on the script : 

![image](https://github.com/pharo-project/pharo/assets/49183340/253dd87b-bcb6-491c-9d73-4e9eeefed2f8)

2 - With a clean image + VM with the trace dll : 

At image start : 

```
[ERROR] 2023-10-09 17:18:27.000 sqMakeMemoryNotExecutableFromTo (extracted/vm/src/win/sqWin32SpurAlloc.c:226):VirtualProtect(x,y,PAGE_EXECUTE_READWRITE): No error
```

DoIt on the script : 

![image](https://github.com/pharo-project/pharo/assets/49183340/d4ae6be5-4a8c-4cda-9d5e-d957d4744978)

```
failed to convert string to wide form: (null)
←[31mIceGenericError: failed to convert string to wide form: url: Param├¿tre incorrect.

←[0mIceLibgitErrorVisitor>>visitGenericError:
IceLibgitErrorVisitor>>visitERROR:
LGit_GIT_ERROR>>acceptError:
[ :error |
                location exists ifTrue: [ location ensureDeleteAll ].
                error acceptError: (IceLibgitErrorVisitor onContext: self) ] in IceGitClone>>execute in Block: [ :error |...
FullBlockClosure(BlockClosure)>>cull:
Context>>evaluateSignal:
Context>>handleSignal:
LGit_GIT_ERROR(Exception)>>pass
[ :error |
                        (error object = LGitErrorTypeEnum giterr_net and: [remainingRetries > 1])
                                ifTrue: [
                                        location ensureDeleteAll.
                                        self cloneRepo: repo cloneOptions: cloneOptions remainingRetries: remainingRetries - 1   ]
                                ifFalse: [ error pass ] ] in IceGitClone>>cloneRepo:cloneOptions:remainingRetries: in Block: [ :error | ...
FullBlockClosure(BlockClosure)>>cull:
Context>>evaluateSignal:
Context>>handleSignal:
LGit_GIT_ERROR(Exception)>>signal
LGit_GIT_ERROR class(LGitCallReturnHandler class)>>signalWith:
LGitReturnCodeEnum>>handleLGitReturnCode
LGitRepository(LGitExternalObject)>>withReturnHandlerDo:
LGitRepository>>clone:options:to:
LGitRepository>>clone:options:
[repo clone: url options: cloneOptions] in IceGitClone>>cloneRepo:cloneOptions:remainingRetries: in Block: [repo clone: url options: cloneOptions]
FullBlockClosure(BlockClosure)>>on:do:
IceGitClone>>cloneRepo:cloneOptions:remainingRetries:
[location ensureCreateDirectory.

        repo := LGitRepository on: location.
        cloneOptions := repo cloneOptionsStructureClass withCredentialsProvider: (IceCredentialsProvider defaultForRemoteUrl: url).

        "Keeping references, because if not the GC take them."
        checkoutOptions := cloneOptions checkoutOptions.
        callbacks := cloneOptions fetchOptions callbacks.
        callbacks transferProgress: IceGitTransferProgress new.

        checkoutOptions checkoutStrategy: LGitCheckoutStrategyEnum git_checkout_force.
        checkoutOptions progressCallback: IceGitCheckoutProgress new.

        self cloneRepo: repo cloneOptions: cloneOptions remainingRetries: 3.

        (LGitRemote of: repo named: 'origin')
                lookup;
                setUrl: url.

        ] in IceGitClone>>execute in Block: [location ensureCreateDirectory....
FullBlockClosure(BlockClosure)>>on:do:
IceGitClone>>execute
IceRepositoryCreator>>cloneRepository
[
                self validate.
                self isCloning
                        ifTrue: [ self cloneRepository ]
                        ifFalse: [ self addLocalRepository ] ] in IceRepositoryCreator>>createRepository in Block: [...
FullBlockClosure(BlockClosure)>>on:do:
IceRepositoryCreator>>createRepository
[ repository := builder createRepository ] in MCGitHubRepository(MCGitBasedNetworkRepository)>>createIcebergRepositoryFor: in Block: [ repository := builder createRepository ]
FullBlockClosure(BlockClosure)>>on:do:
←[0m←[31mIceGenericError: failed to convert string to wide form: url: Param├¿tre incorrect.

←[0mIceLibgitErrorVisitor>>visitGenericError:
IceLibgitErrorVisitor>>visitERROR:
LGit_GIT_ERROR>>acceptError:
[ :error |
                location exists ifTrue: [ location ensureDeleteAll ].
                error acceptError: (IceLibgitErrorVisitor onContext: self) ] in IceGitClone>>execute in Block: [ :error |...
FullBlockClosure(BlockClosure)>>cull:
Context>>evaluateSignal:
Context>>handleSignal:
LGit_GIT_ERROR(Exception)>>pass
[ :error |
                        (error object = LGitErrorTypeEnum giterr_net and: [remainingRetries > 1])
                                ifTrue: [
                                        location ensureDeleteAll.
                                        self cloneRepo: repo cloneOptions: cloneOptions remainingRetries: remainingRetries - 1   ]
                                ifFalse: [ error pass ] ] in IceGitClone>>cloneRepo:cloneOptions:remainingRetries: in Block: [ :error | ...
FullBlockClosure(BlockClosure)>>cull:
Context>>evaluateSignal:
Context>>handleSignal:
LGit_GIT_ERROR(Exception)>>signal
LGit_GIT_ERROR class(LGitCallReturnHandler class)>>signalWith:
LGitReturnCodeEnum>>handleLGitReturnCode
LGitRepository(LGitExternalObject)>>withReturnHandlerDo:
LGitRepository>>clone:options:to:
LGitRepository>>clone:options:
[repo clone: url options: cloneOptions] in IceGitClone>>cloneRepo:cloneOptions:remainingRetries: in Block: [repo clone: url options: cloneOptions]
FullBlockClosure(BlockClosure)>>on:do:
IceGitClone>>cloneRepo:cloneOptions:remainingRetries:
[location ensureCreateDirectory.

        repo := LGitRepository on: location.
        cloneOptions := repo cloneOptionsStructureClass withCredentialsProvider: (IceCredentialsProvider defaultForRemoteUrl: url).
Error printing the compiledBlock in IceGitClone>>execute in Block: [location ensureCreateDirectory....
FullBlockClosure(BlockClosure)>>on:do:
IceGitClone>>execute
IceRepositoryCreator>>cloneRepository
[
                self validate.
                self isCloning
                        ifTrue: [ self cloneRepository ]
                        ifFalse: [ self addLocalRepository ] ] in IceRepositoryCreator>>createRepository in Block: [...
FullBlockClosure(BlockClosure)>>on:do:
IceRepositoryCreator>>createRepository
[ repository := builder createRepository ] in MCGitHubRepository(MCGitBasedNetworkRepository)>>createIcebergRepositoryFor: in Block: [ repository := builder createRepository ]
FullBlockClosure(BlockClosure)>>on:do:
←[0m
```

--- a/src/BaselineOfPharo/BaselineOfPharo.class.st
+++ b/src/BaselineOfPharo/BaselineOfPharo.class.st
@@ -43,7 +43,7 @@ BaselineOfPharo class >> documentBrowser [
 		newName: 'DocumentBrowser' 
 		owner: 'pharo-spec' 
 		project: 'NewTools-DocumentBrowser' 
-		version: 'v1.0.1'
+		version: 'v1.0.2'
 ]
 
 { #category : #'repository urls' }

